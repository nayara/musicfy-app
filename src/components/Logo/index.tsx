import { FC } from "react";
import * as Style from "./Logo.styles";
import { useNavigate } from "react-router-dom";
import { rootPath } from "../../routes";

const Logo: FC = () => {
  const navigate = useNavigate();

  return (
    <Style.Button onClick={() => navigate(rootPath)}>
      <Style.Logo>MusicFy</Style.Logo>
    </Style.Button>
  );
};

export default Logo;
