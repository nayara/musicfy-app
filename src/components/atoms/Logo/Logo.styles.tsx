import { styled } from "styled-components";

const Button = styled.button`
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  width: 50%;
  text-align: left;
`;

const Logo = styled.h1`
  font-family: "Genty";
  font-size: 3vh;
  padding: 1.5vh;
  margin: 0;
`;

export { Logo, Button };
