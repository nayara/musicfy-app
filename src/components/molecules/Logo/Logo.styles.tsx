import { styled } from "styled-components";

const Logo = styled.h1`
  font-family: "Genty";
  font-size: 3vh;
  padding: 1.5vh;
  margin: 0;
`;

export { Logo };
