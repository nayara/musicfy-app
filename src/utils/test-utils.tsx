import { RenderOptions, render } from "@testing-library/react";
import { FC, ReactElement, ReactNode } from "react";
import { Providers } from "src/providers";

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <Providers>{children}</Providers>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
