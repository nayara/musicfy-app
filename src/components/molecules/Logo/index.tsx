import { FC } from "react";
import * as Style from "./Logo.styles";
import { useNavigate } from "react-router-dom";
import { rootPath } from "../../../routes";
import Button from "../../atoms/Button";

const Logo: FC = () => {
  const navigate = useNavigate();

  return (
    <Button onClick={() => navigate(rootPath)}>
      <Style.Logo>MusicFy</Style.Logo>
    </Button>
  );
};

export default Logo;
