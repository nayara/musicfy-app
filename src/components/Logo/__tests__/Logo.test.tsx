import { render, screen } from "../../../utils/test-utils";
import { userEvent } from "@testing-library/user-event";
import Logo from "..";

const mockUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  useNavigate: () => mockUseNavigate,
}));

describe("<Logo>", () => {
  it("should render logo correctly", () => {
    const { container } = render(<Logo />);

    expect(container).toMatchSnapshot();
  });

  it("should redirect to rootPath when clicked", async () => {
    render(<Logo />);

    const logoButton = screen.getByRole("button", { name: "MusicFy" });
    await userEvent.click(logoButton);

    expect(mockUseNavigate).toHaveBeenCalled();
  });
});
