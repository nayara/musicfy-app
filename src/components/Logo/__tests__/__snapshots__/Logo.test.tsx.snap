// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo> should render logo correctly 1`] = `
.c0 {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
}

.c1 {
  font-family: "Genty";
  padding: 1rem;
  margin: 0;
}

<div>
  <button
    class="c0"
  >
    <h1
      class="c1"
    >
      MusicFy
    </h1>
  </button>
</div>
`;
